internal/raft/message/convert.go

```
@@ -0,0 +1,1089 @@
+}
+
+// ConvertCommandToMessageListDistinct converts a command.Distinct to a message.List_Distinct.
+func ConvertCommandToMessageListDistinct(cmd *command.Distinct) *List_Distinct {
+	msgListDistinct := &List_Distinct{
+		&Command_Distinct{
+			Input: ConvertCommandToMessageList(cmd.Input),
+		},
+	}
+	return msgListDistinct
+}
+
+// ConvertCommandToMessageRepeatedExprSlice converts a [][]command.Expr to a [][]message.Expr.
+func ConvertCommandToMessageRepeatedExprSlice(cmd [][]command.Expr) []*RepeatedExpr {
+	msgRepeatedExprSlice := []*RepeatedExpr{}
+	for i := range cmd {
+		msgRepeatedExpr := &RepeatedExpr{}
+		for j := range cmd[i] {
+			msgRepeatedExpr.Expr = append(msgRepeatedExpr.Expr, ConvertCommandToMessageExpr(cmd[i][j]))
+		}
+		msgRepeatedExprSlice = append(msgRepeatedExprSlice, msgRepeatedExpr)
+	}
+	return msgRepeatedExprSlice
+}
+
+// ConvertCommandToMessageListValues converts a command.Values to a message.List_Values.
+func ConvertCommandToMessageListValues(cmd *command.Values) *List_Values {
+	msgListValues := &List_Values{
+		&Command_Values{
+			Expr: ConvertCommandToMessageRepeatedExprSlice(cmd.Values),
+		},
+	}
+	return msgListValues
+}
+
+// ConvertCommandToMessageList converts
+func ConvertCommandToMessageList(cmd command.List) *List {
+	msgList := &List{}
+	switch c := cmd.(type) {
+	case *command.Scan:
+		msgList.List = ConvertCommandToMessageListScan(c)
+	case *command.Select:
+		msgList.List = ConvertCommandToMessageListSelect(c)
+	case *command.Project:
+		msgList.List = ConvertCommandToMessageListProject(c)
+	case *command.Join:
+		msgList.List = ConvertCommandToMessageListJoin(c)
+	case *command.Limit:
+		msgList.List = ConvertCommandToMessageListLimit(c)
+	case *command.Offset:
+		msgList.List = ConvertCommandToMessageListOffset(c)
+	case *command.Distinct:
+		msgList.List = ConvertCommandToMessageListDistinct(c)
+	case *command.Values:
+		msgList.List = ConvertCommandToMessageListValues(c)
+	}
+	return msgList
```

Zero-value when unhandled. Please return an error here